### ISSUE TITLE
[rust 1.32] compilation fails "SIGSEGV: invalid memory reference"

### ISSUE THREAD

**Issue by xenopeek** [dbrgn/tealdeer#76]
OS: Arch Linux
rust: 1.32.0

Tried to build the AUR packages `tealdeer` and `tealdeer-git` and both builds fail with a "SIGSEGV: invalid memory reference" error while compiling tealdeer source, after all the crates were successfully compiled before that point. Output of makepkg on `tealdeer-git` in [tealdeer-git-aur.txt](https://github.com/dbrgn/tealdeer/files/2775515/tealdeer-git-aur.txt).

I also tried manually with `git clone https://github.com/dbrgn/tealdeer` and `cargo build --release --verbose` and that fails with the same error. Output of that in [tealdeer-git-clone.txt](https://github.com/dbrgn/tealdeer/files/2775516/tealdeer-git-clone.txt).

I can't tell if this is a problem on my end, with my rust version or with the tealdeer source. Any help would be appreciated.

**Comment by dbrgn** [comment]
That sounds like a Rust bug to me. Are you using the stable version of Rust? Can you try another version (e.g. Rust 1.31 or the nightly version)?

For me, compilation works on rustc 1.32.0 (9fda7c223 2019-01-16).

**Comment by ranfdev** [comment]
same error as @xenopeek, and same setup. Arch linux, latest rust stable 1.32.0

**Comment by xenopeek** [comment]
It solved itself after removing and reinstalling rust. I have no explanation for it.

I had installed rust from the Arch Linux repository. Removed it and installed nightly through rustup. Build was now successful. Switched to stable through rustup. Build again successful. Reinstalled rust from the Arch Linux repository. Build still successful? Can't pin a cause but solved anyway.

Thanks for the feedback. I'll retest with other rust versions when next I have some issue like this.
